### PR TITLE
Rspec for user login

### PR DESCRIPTION
**Pull request by alexvis**

### CODE REVIEW

**Comment by nwalberts** [comment]
Please add:
At the top of your Rubocop file should be 
```
AllCops:
  Exclude:
    - db/schema.rb
    - react
```
So as to keep Hound away from your JS code

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class UsersController < ApplicationController
+  def index; end
+
+  def create; end
+
+  def show; end
+end
```
Comment (nwalberts) [inline app/controllers/users_controller.rb]: please remove these methods if you are not going to use them

File: app/views/index/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<div id="app">
+</div>
+
+<h1>hello alex<h1>
```
Comment (nwalberts) [inline app/views/index/index.html.erb]: this confuses me. There should not be an `index` folder.  Im not even sure what would render this? Additionally, you would want to remove the `hello alex`

Instead, this should maybe be in a static_pages folder, and you should have a specifc action in your controller refer to it. Lots of different ways to implement it, but right now Im not sure what this does

File: spec/features/sign_up_spec.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
+require 'rails_helper'
+
```
Comment (nwalberts) [inline spec/features/sign_up_spec.rb:2]: please remove this user story text

File: spec/features/user_signs_in_spec.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  scenario 'an existing user specifies a valid email and password' do
+    user = FactoryGirl.create(:user)
+    visit root_path
+    click_link 'Sign In'
+    fill_in 'Email', with: user.email
+    fill_in 'Password', with: user.password
+    click_button 'Sign In'
+    expect(page).to have_content('Welcome Back!')
+    expect(page).to have_content('Sign Out')
+  end
```
Comment (nwalberts) [inline spec/features/user_signs_in_spec.rb]: Please add spacing

File: spec/features/user_signs_in_spec.rb [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    fill_in 'Password', with: user.password
+    click_button 'Sign In'
+
+    expect(page).to have_content('Sign Out')
+    expect(page).to_not have_content('Sign In')
+
+    visit new_user_session_path
+    expect(page).to have_content('You are already signed in.')
+  end
+end
```
Comment (nwalberts) [inline spec/features/user_signs_in_spec.rb]: please add spacing between scenario blocks

File: spec/features/user_signs_in_spec.rb [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    expect(page).to have_content('Invalid Email or password.')
+    expect(page).to_not have_content('Sign Out')
+  end
+  scenario 'an already authenticated user cannont re-sign in' do
+    user = FactoryGirl.create(:user)
+    visit new_user_session_path
+    fill_in 'Email', with: user.email
+    fill_in 'Password', with: user.password
+    click_button 'Sign In'
+
```
Comment (nwalberts) [inline spec/features/user_signs_in_spec.rb:48]: I would strongly suggest using your own helper method or a devise helper method such as `login_as` in order to skip the sign in process for these tests 
https://github.com/plataformatec/devise/wiki/How-To:-Test-with-Capybara

File: react/src/App.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+
+const App = (props) => {
+  return(
+     <div>
+     <h1>hello from App</h1>
+     </div>
+ )
+}
+
```
Comment (nwalberts) [inline react/src/App.js]: Fix your indentation across this page. 
and you are importing Router elements that are not used. Delete line 2

File: react/src/components/NavBar.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import React from 'react'
+import Link from 'react-router'
+
+
+
+const NavBar = (props) => {
+  return
+  }
+
+export default NavBar;
```
Comment (nwalberts) [inline react/src/components/NavBar.js]: Delete this file, you are not using it in this PR

File: app/controllers/users_controller.rb [hunk 9]
```diff
@@ -0,0 +1,3 @@
+class UsersController < ApplicationController
+  def index; end
```
Comment (amandabeiner) [inline app/controllers/users_controller.rb:2]: I might personally just make this a regularly formatted `index` block with nothing in it for readability/convention :)

File: spec/features/user_signs_in_spec.rb [hunk 10]
```diff
@@ -0,0 +1,56 @@
+require 'rails_helper'
+
+feature 'user signs in', %Q{
+  As a user
+  I want to sign in
+  So that I can track my park activities
+} do
+
+  scenario 'an existing user specifies a valid email and password' do
+    user = FactoryGirl.create(:user)
```
Comment (amandabeiner) [inline spec/features/user_signs_in_spec.rb:10]: Love the use of FactoryGirl!

File: spec/features/user_signs_in_spec.rb (hunk 6 above)
Comment (amandabeiner) [reply spec/features/user_signs_in_spec.rb:48]: Totes agree with Nick here! You could also extract this granular sign in instructions into a `before(:each)` block!

File: spec/models/user_spec.rb [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+
+  it 'has a matching password confirmation for the password' do
+    user = User.new
+    user.password = 'password'
+    user.password_confirmation = 'whatever'
+
+    expect(user).to_not be_valid
+    expect(user.errors[:password_confirmation]).to_not be_blank
+  end
+
```
Comment (amandabeiner) [inline spec/models/user_spec.rb:27]: Really nice model test! Don't forget to come back here and update it as you add associations to the `user` table. So, for instance, if a user `has_many` `reviews` in the future, you'll want to come back to this model spec and test for that new association! 

(Not that you should be doing this right now, just a heads up for the future)

File: react/webpack.config.js [hunk 12]
```diff
@@ -0,0 +1,29 @@
+const path = require('path');
```
[the remark this answers is not in the document]
Comment (amandabeiner) [reply react/webpack.config.js:1]: Make sure you update hound to acount for ES6 if you want to use it for JavaScript!

**Pull request merged**